File: R/7kyu/7kyu-square-every-digit.r
# Square Every Digit
# Langugage: R
# Difficulty: 7 kyu

# Solution 1
square_digits <- function(num){
  split_num = as.numeric(strsplit(as.character(num), "")[[1]])
  as.integer(paste(split_num^2, collapse = ""))
}

# One line of Solution 1
square_digits <- function(num){
  as.numeric(paste((as.character(as.numeric(strsplit(as.character(num),"")[[1]])^2)), collapse = ""))
}


# Tests
test_that('Basic tests', {
  expect_equal(square_digits(3212), 9414)
  expect_equal(square_digits(2112), 4114)
  expect_equal(square_digits(0), 0)
})

File: Python/7kyu/7kyu-square-every-digit.py
# Square Every Digit
# Langugage: Python
# Difficulty: 7 kyu

# Solution 1
def square_digits(num):
    split_num = [int(x) for x in str(num)]
    result = ''
    for n in split_num:
        result += str(n ** 2)
    return int(result)

# More concise alternative to Solution 1
def square_digits(num):
    result = ''
    for x in str(num):
        result += str(int(x) ** 2)
    return int(result)

# Tests
import codewars_test as test
from solution import square_digits

@test.describe("Premade tests: ")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(square_digits(9119), 811181)
        test.assert_equals(square_digits(0), 0)